lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -4593,21 +4595,39 @@ mkApiCoinSelection deps refunds mcerts metadata unsignedTx =
             <$> unsignedTx ^. #unsignedCollateral
         , withdrawals = mkApiCoinSelectionWithdrawal
             <$> unsignedTx ^. #unsignedWithdrawals
-        , certificates = uncurry mkApiCoinSelectionCerts
-            <$> mcerts
+        , certificates =
+            mergeCerts delCertsM votingCertsM
         , depositsTaken = ApiAmount.fromCoin
             <$> deps
         , depositsReturned = ApiAmount.fromCoin
             <$> refunds
         , metadata = ApiBytesT. serialiseToCBOR
             <$> metadata
         }
-
-mkApiCoinSelectionCerts
+  where
+    delCertsM = uncurry mkApiCoinSelectionDelCerts <$> mDelCerts
+    votingCertsM = uncurry mkApiCoinSelectionVotingCerts <$> mVotingCerts
+    mergeCerts del vote = case (del, vote) of
```

Comment:
I would really want to have tests for this :)

Reply:
And I mean unit test. I imagine the various integration voting tests are there to cover various cases, but it woiuld be much more efficient and clear to cover those cases in a unit test dedicated to the merge function, and have only _one_ integration test that ensures this endpoint is hit

Reply:
in the end, I have come to conclusion that we should present deleg/voting certs as they are (this is constructTransaction), plus we want the same result as in the case of decodeTransaction. So `mergeCerts` is just combined via `semigroup <>`. So no logic really.